src/jit/compiler.hpp

```diff
@@ -4719,9 +4734,9 @@ inline bool         BasicBlock::endsWithJmpMethod(Compiler *comp)
 {
     if (comp->compJmpOpUsed && (bbJumpKind == BBJ_RETURN) && (bbFlags & BBF_HAS_JMP))
     {
-        GenTreePtr last = comp->fgGetLastTopLevelStmt(this);
-        assert(last != nullptr);
-        return last->gtStmt.gtStmtExpr->gtOper == GT_JMP;
+        GenTree* lastNode = IsLIR() ? bbLastNode : comp->fgGetLastTopLevelStmt(this)->gtStmt.gtStmtExpr;
```

Comment:
>bbLastNode [](start = 38, length = 10)

I wonder if we should have LIRFirstNode() / LIRLastNode() accessors. This would help, as you could assert the block is in LIR form (probably with a new BBF_* flag) when the accessors are used. There could be corresponding setters.